### ISSUE TITLE
[BUG] Azure.Messaging.EventGrid SDK does not parse KeyVault system event data types correctly due to case mismatch

### ISSUE THREAD

**Issue by AtOMiCNebula** [Azure/azure-sdk-for-net#22932]
**Describe the bug**
I've been trying to work with some KeyVault events, and have found that using the new `Azure.Messaging.EventGrid` SDK to read the events produces blank objects.  The object instance is the correct type, but all properties are null.  As best as I can tell, this is because the delivered events use a different casing than the SDK expects.  For instance, here is a (slightly cleaned up) event instance:

```json
{
    "id": "39869b5c-6d56-404b-aab8-50ec200f93f8",
    "topic": "/subscriptions/***/resourceGroups/sample-rg/providers/Microsoft.KeyVault/vaults/sample-vault",
    "subject": "TestSecret",
    "eventType": "Microsoft.KeyVault.SecretNewVersionCreated",
    "data": {
        "Id": "https://sample-vault.vault.azure.net/secrets/TestSecret/a9f4eab27ec64771ac1fb07d38fdbae2",
        "VaultName": "sample-vault",
        "ObjectType": "Secret",
        "ObjectName": "TestSecret",
        "Version": "a9f4eab27ec64771ac1fb07d38fdbae2",
        "NBF": 1627491993,
        "EXP": 1690563970
    },
    "dataVersion": "1",
    "metadataVersion": "1",
    "eventTime": "2021-07-28T17:08:38.2145376Z"
}
```

The casing on the data properties is title-cased (documented as such [here](https://docs.microsoft.com/en-us/azure/event-grid/event-schema-key-vault?tabs=event-grid-event-schema#event-examples)), but the SDK expects camel-casing (as code-generated [here](https://github.com/Azure/azure-sdk-for-net/blob/fb1c53b9a49af7241f738b95b50e4e55b56fbb92/sdk/eventgrid/Azure.Messaging.EventGrid/src/Generated/Models/KeyVaultSecretNewVersionCreatedEventData.Serialization.cs#L29)).  The `Microsoft.Azure.EventGrid` SDK appears to have used this incorrect casing too ([here](https://github.com/Azure/azure-sdk-for-net/blob/fb1c53b9a49af7241f738b95b50e4e55b56fbb92/sdk/eventgrid/Microsoft.Azure.EventGrid/src/Generated/Models/KeyVaultSecretNewVersionCreatedEventData.cs#L69)), but it still produced working event data instances since NSJ fuzzy-matched on the property keys.  I can get the object to parse correctly if I modify the payload to have the expected casing, but that's not something I'd want to deploy.

There may be other event data structures beyond KeyVault types that have this inconsistency as well, but I haven't looked too closely.  For now, I'm moving back to the previous-generation SDK so that I can be unblocked here, but this seems like a pretty glaring bug in the new SDK.  Hopefully I can update back to it at some point. :smile:

**Expected behavior**
`Azure.Messaging.EventGrid` parses KeyVault SecretNewVersion (and other) event data structures correctly.

**Actual behavior (include Exception or Stack Trace)**
An instance of the correct type is provided, but it has all null values.

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Create a Key Vault.
2. Register an event subscription for New Secret Version events.
3. Parse the provided payload with Azure.Messaging.EventGrid.

**Environment:**
 - Name and version of the Library package used: Azure.Messaging.EventGrid 4.5.0
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): .NET Core 3.1 and 5.0
 - IDE and version : Visual Studio 16.10.4

**Comment by JoshLove-msft** [comment]
Hi @AtOMiCNebula,
We are sorry that you are running into this issue. The [swagger](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/eventgrid/data-plane/Microsoft.KeyVault/stable/2018-01-01/KeyVault.json#L10) defines the payloads of the event as using camelCase, which is why the SDK expects camelCase. We will need to reach out to the team that is producing the events to understand what is going wrong here.

**Comment by AtOMiCNebula** [comment]
Ah, interesting.  My money's on "the swagger is wrong, oops", since changing the content the API returns at this point might be considered a breaking change. :innocent:

**Comment by JoshLove-msft** [comment]
This is fixed in 4.6.0 - https://www.nuget.org/packages/Azure.Messaging.EventGrid/4.6.0

**Comment by AtOMiCNebula** [comment]
Thanks, #23208 / v4.6.0 works!
